src/components/Form/Form.tsx

```diff
@@ -66,22 +157,349 @@ export default function Form({ setFormData }: IFormProps) {
+                    size="small"
+                    onChange={onRateInputChange}
+                    onBlur={handleBlur}
+                    inputProps={{
+                      step: 1,
+                      min: 0,
+                      max: 100,
+                      type: "number",
+                      "aria-labelledby": "input-slider",
+                    }}
+                    sx={{
+                      mt: -1,
+                      "& .MuiInputBase-input": {
+                        color: "#213547", // Text color
+                        backgroundColor: "#f9f9f9", // Background color
+                        borderRadius: "4px", // Border radius
+                        padding: "0.5em", // Padding
+                        border: "1px solid #ddd", // Border
+                        "&:hover": {
+                          borderColor: "#747bff", // Border color on hover
+                        },
+                        "&.Mui-focused": {
+                          borderColor: "#747bff", // Border color when focused
+                        },
+                      },
+                    }}
+                  />
+                </Grid>
+              </Grid>
+            </Box>
+          </div>
+          <Stack direction="row" spacing={2}>
+            <Button
+              variant="contained"
+              endIcon={<SendIcon />}
+              onClick={handleSubmit}
+              sx={{
+                borderRadius: "8px",
+                border: "1px solid transparent",
+                padding: "0.6em 1.2em",
+                fontSize: "1em",
+                fontWeight: "500",
+                fontFamily: "inherit",
+                backgroundColor: "#1a1a1a",
+                cursor: "pointer",
+                transition: "border-color 0.25s",
+                "&:hover": {
+                  borderColor: "#646cff",
+                },
+                "&:focus": {
+                  outline: "4px auto -webkit-focus-ring-color",
+                },
+              }}
+            >
+              Submit
+            </Button>
+            <Button
+              variant="outlined"
+              startIcon={<ClearIcon />}
+              onClick={handleClear}
+              sx={{
+                borderRadius: "8px",
+                border: "1px solid transparent",
+                padding: "0.6em 1.2em",
+                fontSize: "1em",
+                fontWeight: "500",
+                fontFamily: "inherit",
+                cursor: "pointer",
+                transition: "border-color 0.25s",
+                "&:hover": {
+                  borderColor: "#646cff",
+                },
+                "&:focus": {
+                  outline: "4px auto -webkit-focus-ring-color",
+                },
+              }}
+            >
+              Clear
+            </Button>
+          </Stack>
         </section>
       </form>
     </div>
   );
-}
+};
+
+export default Form;
```

Comment:
505 lines of code a bit too much and looks really suspicious, please try to move reusable parts to different files and `Form.tsx` be a bit smaller and more developer friendly